master_discovery_fkie/src/master_discovery_fkie/master_discovery.py

```diff
@@ -1089,3 +1119,26 @@ def rosservice_refresh(self, req):
                 import traceback
                 traceback.print_exc()
         return []
+
+
+    def log_message_counters(self):
+        with self.__lock:
+            try:
+                current = self.msg_counter_sent_group
+                delta = current - self.msg_counter_sent_group_last
```

Comment:
Noting that `delta` in this method (both uses) can be negative; reporting this is fine. Is this what you wanted?